### PR TITLE
Remove ability to sort in SNPs index

### PR DESCRIPTION
**Pull request by philippbayer**

### CODE REVIEW

**Comment by philippbayer** [comment]
Hmm problematical: you can't press the buttons but the URL parameters like '/snps?direction=asc&page=2&sort=chromosome' are still accepted and work, how do I turn that off?

**Comment by gedankenstuecke** [comment]
You have to remove the parameters from the controllers for that I think :)

File: app/views/snps/index.html.erb [hunk 1]
```diff
@@ -10,16 +10,16 @@
       <p class="text-center snps__text-download">Includes annotation for all SNPs from all sources</p>
     </div>
   </div>
-  <div class="table-responsive">
```
Comment (gedankenstuecke) [inline app/views/snps/index.html.erb]: Don't know whether the comment made it: but it should stay responsive as it's about the design for mobiles/tablets

**Comment by philippbayer** [comment]
I 'solved' it in an ugly way now. The responsive part has returned

File: app/controllers/snps_controller.rb [hunk 2]
```diff
@@ -4,6 +4,8 @@ class SnpsController < ApplicationController
   before_filter :find_snp, except: [:index, :json,:json_annotation]
 
   def index
+    params.delete(:sort)
+    params.delete(:direction)
     @snps = Snp.order(sort_column + ' ' + sort_direction)
```
Comment (gedankenstuecke) [inline app/controllers/snps_controller.rb]: Wouldn't it have been enough to set `Snp.order(ranking: :desk)`, [analogous to this](https://apidock.com/rails/ActiveRecord/QueryMethods/order)? Then the additional parameters wouldn't matter at all?

**Comment by gedankenstuecke** [comment]
Ok, this should make for a nicer fix of the parameters than just deleting them from the param's hash. @tsujigiri I think it's good to go. what do you think?

**Comment by philippbayer** [comment]
Good idea to get rid of the sort_column etc. stuff, it's from *2014*

**Comment by gedankenstuecke** [comment]
Ah, you're still awake, had assumed you went to 💤 ! Looks good to me now.

**Comment by philippbayer** [comment]
Ah et's give it a go, maybe the server will stay up while I sleep :)

File: app/controllers/snps_controller.rb [hunk 3]
```diff
@@ -1,10 +1,11 @@
 # frozen_string_literal: true
 class SnpsController < ApplicationController
-  helper_method :sort_column, :sort_direction
   before_filter :find_snp, except: [:index, :json,:json_annotation]
 
   def index
-    @snps = Snp.order(sort_column + ' ' + sort_direction)
+    params.delete(:sort)
+    params.delete(:direction)
```
Comment (tsujigiri) [inline app/controllers/snps_controller.rb]: Deleting the params is likely not necessary. By not reading the params, you are already ignoring them.

Comment (philippbayer) [reply app/controllers/snps_controller.rb]: I thought the same! But playing around with it on my local system the table kept on getting sorted depending on the params in the URL I set manually, I didn't think that was possible

Comment (tsujigiri) [reply app/controllers/snps_controller.rb]: Me neither... :thinking:

Comment (philippbayer) [reply app/controllers/snps_controller.rb]: it is a superugly solution and I welcome better ideas :D

Comment (gedankenstuecke) [reply app/controllers/snps_controller.rb]: Wait, why are these back? With my change to `Snp.order()` this really isn't needed as the parameters are a) never parsed as the helper methods are gone and b) the order command now directly states that the ordering is performed on ranking?

Comment (gedankenstuecke) [reply app/controllers/snps_controller.rb]: You had to delete them earlier as you didn't change the `order` call, but my commits in this PR changed that ☺️

Comment (philippbayer) [reply app/controllers/snps_controller.rb]: Hmm i didn't put them back in, must've been github merging commits?

**Pull request merged**